app/addons/documents/doc-editor/components.react.jsx

```diff
@@ -218,44 +218,137 @@ var AttachmentsPanelButton = React.createClass({
     };
   },
 
+  getInitialState: function () {
+    return {
+      filter: '',
```

Comment:
I think this filter should be kept in the store and the list of documents is retrieved from the store. So that the `getAttachmentsList` function goes back to just being a simple `map` function.